File: java/src/phase 1-conditional_thinking_java/Q39ElectricityBill.java
import java.util.Scanner;

/**
 * Phase 1: Conditional Thinking (If-Else, Boolean Logic)
 * Level 4: Logical Operators & Compound Statements
 * Question 39: Take electricity units consumed and calculate the bill as per slabs (using if-else).
 * 
 * Mentor Tips (Java 26):
 * - Define slabs (e.g., first 100 units at $1/unit, next 200 at $2/unit, above that $5/unit). Calculate progressively.
 */
public class Q39ElectricityBill {

    public static double solve(double units) {
        // TODO: Implement this method to solve the question
        0.0;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter units: "); double u = scanner.nextDouble();
                System.out.println("Bill: " + solve(u));
        scanner.close();
    }
}

File: python/phase 1-condtional_thinking/q39_electricity_bill.py
"""
Phase 1: Conditional Thinking (If-Else, Boolean Logic)
Level 4: Logical Operators & Compound Statements
Question 39: Take electricity units consumed and calculate the bill as per slabs (using if-else).

Mentor Tips (Python 3.14+):
- Define slabs (e.g., first 100 units at $1/unit, next 200 at $2/unit, above that $5/unit). Calculate progressively.
"""

def solve(units: float) -> float:
    # TODO: Implement this method to solve the question
    if units <= 100:
        return units * 1.0
    elif units <= 300:
        return (100 * 1.0) + ((units - 100) * 2.0)
    else:
        return (100 * 1.0) + (200 * 2.0) + ((units - 300) * 5.0)
    return 0.0

def main():
    units = float(input("Enter units: "))
    print(f"Bill: {solve(units)}")

if __name__ == "__main__":
    main()
